### PR TITLE
Enable ETAG Headers for Static Files

### PR DESCRIPTION
**Pull request by S3bb1**
This PR enables ETAG Header generation for static served files

### CODE REVIEW

File: servers/web.js [hunk 1]
```diff
@@ -105,12 +107,21 @@ var initialize = function(api, options, next){
...
     var headers = connection.rawConnection.responseHeaders;
     if(error){ connection.rawConnection.responseHttpCode = 404; }
     if(ifModifiedSince && lastModified <= ifModifiedSince){ connection.rawConnection.responseHttpCode = 304; }
+    if(api.config.servers.web.enableEtag && fileStream){
+      var fileBuffer = !Buffer.isBuffer(fileStream) ? new Buffer(fileStream.toString(), 'utf8') : fileStream;
+      var fileEtag = etag(fileBuffer, {weak: true});
+      connection.rawConnection.responseHeaders.push(['ETag', fileEtag]);
+      var lastModifiedDate = new Date(lastModified);
+      lastModifiedDate.setMilliseconds(0);
+      if(fresh(reqHeaders, {'etag': fileEtag, 'last-modified':lastModifiedDate})){
```
Comment (evantahler) [inline servers/web.js]: Why do you need a whole package (`fresh`) here?  Aren't you just comparing the requests' `if-none-match` header to the new etag?

File: servers/web.js [hunk 2]
```diff
@@ -105,12 +107,21 @@ var initialize = function(api, options, next){
...
+        connection.rawConnection.responseHttpCode = 304;
```
Comment (evantahler) [inline servers/web.js]: If you do have a header match, I think you are just supposed to respond with 304; and not send the body payload.  I think you still will here

File: test/core/staticFile.js [hunk 3]
```diff
@@ -72,6 +72,27 @@ describe('Core: Static File', function(){
...
+      var etag = response.headers['etag'];
+      var options = {
+        url: url + '/simple.html',
+        headers: {
+          'If-None-Match': etag
+        },
+        method: 'get'
+      };
+      request(options, function(error, response){
+        response.statusCode.should.equal(304);
```
Comment (evantahler) [inline test/core/staticFile.js]: test should ensure that there is no body returned

**Comment by evantahler** [comment]
awesome start!

File: test/core/staticFile.js (hunk 3 above)
Comment (S3bb1) [inline test/core/staticFile.js]: fixed it

File: servers/web.js (hunk 2 above)
Comment (S3bb1) [inline servers/web.js]: No, in the line above it is handled the same. and the check after the logic checks if the responsecode is !== 304 :)

File: servers/web.js (hunk 1 above)
Comment (S3bb1) [inline servers/web.js]: yeah thats right. I've removed the module and added the check directly

**Comment by evantahler** [comment]
💪

**Pull request merged**